modules/packages/HDF5_HL.chpl

```diff
@@ -3681,6 +3681,149 @@ module HDF5_HL {
       }
     }
 
+    /* Read the dataset named `dset` from the HDF5 file named `filename`.
+       The dataset consists of elements of type `eltType`.  The file will
+       be read in chunks matching the `chunkShape` domain, and yielded as
+       arrays of that size until the end of the dataset is reached.
+
+       Currently, the `chunkShape` domain describing the output arrays and
+       the shape of the data in the file must both be the same rank.
```

Comment:
I suggest adding something like "For example, if the data in the file is 2D, `chunkShape` must be 2-dimensional, too."